lib/subordinateio.c

```diff
@@ -311,31 +311,31 @@ static bool append_range(struct subordinate_range ***ranges, const struct subord
 {
 	struct subordinate_range *tmp;
 	if (!*ranges) {
-		*ranges = malloc(2 * sizeof(struct subordinate_range **));
+		*ranges = malloc(sizeof(struct subordinate_range **));
```

Comment:
Why is this `sizeof(struct subordinate_range **)` and not `sizeof(struct subordinate_range *)`?

Result is clearly the same. I'm just confused.

Reply:
As you say both pointer types are the same size , but technically because ranges is a ***, *range is a **.

Reply:
*boggle* i had that wrong.  But in any case I'm changing it.

Reply:
I don't object keeping this as an array of pointers to a struct, but still this should be `sizeof(struct subordinate_range *)` :)

Reply:
Right.